cpp/test/core/equal.h

```diff
@@ -35,110 +103,120 @@ class Compare
     void operator()(const Field&);
 
 private:
-    Compare& operator=(const Compare&);
+    Comparer& operator=(const Comparer&);
 
     const T& left;
     const T& right;
     bool&    equal;
 };
 
 
-inline bool Equal(double left, double right)
+template <typename Protocols = bond::BuiltInProtocols, typename T1, typename T2>
+bool Compare(const T1& lhs, const T2& rhs)
+{
+    return lhs == rhs;
+}
+
+
+template <typename Protocols = bond::BuiltInProtocols, typename T>
```

Comment:
From what I understand, the `Compare` (former `operator==`) is doing a simple value to value comparison, with some custom logic in some structs, while `Equal` compares the structures being tested in a way test is concerned (e.g. deserialize a `bonded<T>` instead of calling `bonded<T>::operator==`).